apps/webapp/app/routes/_app.orgs.$organizationSlug.projects.$projectParam.runs/route.tsx

```diff
@@ -48,6 +110,70 @@ export default function Page() {
   const isLoading = navigation.state !== "idle";
   const organization = useOrganization();
   const project = useProject();
+  const user = useUser();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const url = new URLSearchParams(location.search);
+
+  const [selectedEnvironment, setSelectedEnvironment] = useState<string>(
+    ExtendedRuntimeEnvironment.ALL
+  );
+  const [selectedStatus, setselectedStatus] = useState<string>(ExtendedJobRunStatus.ALL);
+  const [filteredList, setFilteredList] = useState(list.runs);
+
+  const handleFilterChange = (filterType: string, value: string) => {
+    url.set(filterType, value);
+
+    const filters = ["status", "environment"]; // Add more filters as needed
+    const queryString = filters
+      .map((filter) => (url.has(filter) ? `${filter}=${url.get(filter)}` : null))
+      .filter((filter) => filter !== null)
+      .join("&");
+
+    navigate(`${location.pathname}?${queryString}`);
+  };
+
+  const handleStatusChange = (value: FilterableStatus) => {
+    handleFilterChange("status", value);
+    setselectedStatus(value);
+  };
+
+  const handleEnvironmentChange = (value: string) => {
+    handleFilterChange("environment", value);
+    setSelectedEnvironment(value);
+  };
+
+  useEffect(() => {
+    const status = url.get("status");
+    const environment = url.get("environment");
+
+    const statusArray = filterableStatuses[status as FilterableStatus];
+
+    if (
+      (!status || !statusArray || statusArray.includes("ALL")) &&
+      (!environment || environment === ExtendedRuntimeEnvironment.ALL)
+    ) {
+      setselectedStatus(ExtendedJobRunStatus.ALL);
+      setSelectedEnvironment(ExtendedRuntimeEnvironment.ALL);
+      setFilteredList(list.runs);
+      return;
+    }
+
+    // Filter based on both status and environment
+    let filteredRuns = list.runs;
+
+    if (status && statusArray && !statusArray.includes("ALL")) {
+      filteredRuns = filteredRuns.filter((run) => statusArray.includes(run.status));
+      setselectedStatus(status);
+    }
+
+    if (environment && environment !== ExtendedRuntimeEnvironment.ALL) {
+      filteredRuns = filteredRuns.filter((run) => run.environment.type === environment);
+      setSelectedEnvironment(environment);
+    }
+
+    setFilteredList(filteredRuns);
+  }, [location.search, list.runs, setFilteredList, setselectedStatus, setSelectedEnvironment]);
```

Comment:
This is client-side filtering which isn't the right choice for this. You can have millions of runs which we can't load on the client, so the filtering needs to happen in `RunListPresenter` by passing in the filter data and then incorporating that filter into the database query.